app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt

```diff
@@ -0,0 +1,558 @@
+
+        for (position in firstPosition..lastPosition) {
+            val view = this.findViewByPosition(position)
+            if (view != null) {
+                val viewWidth = view.measuredWidth
+                val viewStartX = view.x
+                val viewEndX = viewStartX + viewWidth
+                if (viewStartX < midScreenX && viewEndX > midScreenX) {
+                    return position
+                }
+            }
+        }
+
+        return RecyclerView.NO_POSITION
+    }
+
+    return RecyclerView.NO_POSITION
+}
+
+/**
+ * Get current position for [LinearLayoutManager.VERTICAL]
+ *
+ * @return position of center item if it exists else
+ * @return first completely visible or first visible item position
+ *
+ * @return [RecyclerView.NO_POSITION] if [LinearLayoutManager] doesn't have vertical orientation
+ */
+
+private fun LinearLayoutManager.getCurrentPositionForVerticalOrientation(midScreenY: Int): Int {
+    if (this.orientation == LinearLayoutManager.VERTICAL) {
+        val centerPosition = getCenterPositionForVerticalOrientation(midScreenY)
+        return if (centerPosition == RecyclerView.NO_POSITION) {
+            getCompletelyVisibleOrFirstPosition()
+        } else centerPosition
+    }
+
+    return RecyclerView.NO_POSITION
+}
+
+/**
+ * Get position of center item for [LinearLayoutManager.VERTICAL]
+ *
+ * @return position of item with start Y less than middle of the screen and end Y more than middle of the screen
+ *
+ * @return [RecyclerView.NO_POSITION] if [LinearLayoutManager] doesn't have vertical orientation or there is no item that meets requirements
+ */
+private fun LinearLayoutManager.getCenterPositionForVerticalOrientation(midScreenY: Int) : Int {
+    if (this.orientation == LinearLayoutManager.VERTICAL) {
+        val firstPosition = this.findFirstVisibleItemPosition()
+        val lastPosition = this.findLastVisibleItemPosition()
+
+        for (position in firstPosition..lastPosition) {
+            val view = this.findViewByPosition(position)
+            if (view != null) {
+                val viewHeight = view.measuredHeight
+                val viewStartY = view.y
+                val viewEndY= viewStartY + viewHeight
+                if (viewStartY < midScreenY && viewEndY > midScreenY) {
+                    return position
+                }
+            }
+        }
+
+        return RecyclerView.NO_POSITION
+    }
+
+    return RecyclerView.NO_POSITION
+}
+
+fun View.setBottomPadding(bottomPadding: Int) {
+    setPadding(paddingLeft, paddingTop, paddingRight, bottomPadding)
+}
+
+fun Context.inflate(resource: Int, root: ViewGroup? = null, attachToRoot: Boolean = false): View {
+    return LayoutInflater.from(this).inflate(resource, root, attachToRoot)
+}
+
+fun Context.getColorCompat(@ColorRes color: Int) = ContextCompat.getColor(this, color)
+
+fun Context.getDrawableCompat(@DrawableRes drawable: Int) = ContextCompat.getDrawable(this, drawable)
```

Comment:
Part of Contex extensions ?

Reply:
we must discuss it on android fridays.

Reply:
Don't remember if we discussed it already, but let's follow the original caller for an extension? `ContextCompat` => `ContextExtentions`?